From 1afba9aa98436551db4c3589eed943f49a106293 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@gnome.org>
Date: Mon, 13 Aug 2012 06:17:06 +0000
Subject: ewsclient: Fix breakage because of libxml2 API breakage

 * libxml2 introduced some API breakage recently.
 * Fix the build so it works with both old and new versions
   of libxml2.

https://bugzilla.gnome.org/show_bug.cgi?id=681720
---
diff --git a/src/goabackend/goaewsclient.c b/src/goabackend/goaewsclient.c
index 6627514..30eb600 100644
--- a/src/goabackend/goaewsclient.c
+++ b/src/goabackend/goaewsclient.c
@@ -351,8 +351,13 @@ ews_client_post_restarted_cb (SoupMessage *msg, gpointer data)
   soup_message_set_request(msg,
                            "text/xml; charset=utf-8",
                            SOUP_MEMORY_COPY,
+#ifdef LIBXML2_NEW_BUFFER
+                           (gchar *) xmlOutputBufferGetContent(buf),
+                           xmlOutputBufferGetSize(buf));
+#else
                            (gchar *) buf->buffer->content,
                            buf->buffer->use);
+#endif
 }
 
 static SoupMessage *
@@ -368,8 +373,13 @@ ews_client_create_msg_for_url (const gchar *url, xmlOutputBuffer *buf)
       soup_message_set_request (msg,
                                 "text/xml; charset=utf-8",
                                 SOUP_MEMORY_COPY,
+#ifdef LIBXML2_NEW_BUFFER
+                                (gchar *) xmlOutputBufferGetContent(buf),
+                                xmlOutputBufferGetSize(buf));
+#else
                                 (gchar *) buf->buffer->content,
                                 buf->buffer->use);
+#endif
       g_signal_connect (msg, "restarted", G_CALLBACK (ews_client_post_restarted_cb), buf);
     }
 
--
cgit v0.9.0.2
